pkg/storage/utils/decomposedfs/upload/session.go

```diff
@@ -0,0 +1,269 @@
+	return s.info
+}
+func (s *OcisSession) ProviderID() string {
+	return s.info.MetaData["providerID"]
+}
+func (s *OcisSession) SpaceID() string {
+	return s.info.Storage["SpaceRoot"]
+}
+func (s *OcisSession) NodeID() string {
+	return s.info.Storage["NodeId"]
+}
+func (s *OcisSession) NodeParentID() string {
+	return s.info.Storage["NodeParentId"]
+}
+func (s *OcisSession) NodeExists() bool {
+	return s.info.Storage["NodeExists"] == "true"
+}
+func (s *OcisSession) LockID() string {
+	return s.info.MetaData["lockid"]
+}
+func (s *OcisSession) ETag() string {
+	return s.info.MetaData["etag"]
+}
+func (s *OcisSession) HeaderIfMatch() string {
+	return s.info.MetaData["if-match"]
+}
+func (s *OcisSession) HeaderIfNoneMatch() string {
+	return s.info.MetaData["if-none-match"]
+}
+func (s *OcisSession) HeaderIfUnmodifiedSince() string {
+	return s.info.MetaData["if-unmodified-since"]
+}
+func (s *OcisSession) Node(ctx context.Context) (*node.Node, error) {
+	n, err := node.ReadNode(ctx, s.store.lu, s.SpaceID(), s.info.Storage["NodeId"], false, nil, true)
+	if err != nil {
+		return nil, err
+	}
+	return n, nil
+}
+
+func (s *OcisSession) ID() string {
+	return s.info.ID
+}
+func (s *OcisSession) Filename() string {
+	return s.info.Storage["NodeName"]
+}
+func (s *OcisSession) Chunk() string {
+	return s.info.Storage["Chunk"]
+}
+func (s *OcisSession) SetMetadata(key, value string) {
+	s.info.MetaData[key] = value
+}
+func (s *OcisSession) SetStorageValue(key, value string) {
+	s.info.Storage[key] = value
+}
+func (s *OcisSession) SetSize(size int64) {
+	s.info.Size = size
+}
+func (s *OcisSession) SetSizeIsDeferred(value bool) {
+	s.info.SizeIsDeferred = value
+}
+
+// TODO get rid of dir, whoever consumes the reference should be able to deal with a relative reference
+// Dir is only used to:
+// * fill the Path property when emitting the UploadReady event after postprocessing finished. I wonder why the UploadReady contains a finished flag ... maybe multiple distinct events would make more sense.
+// * build the reference that is passed to the FileUploaded event in the UploadFinishedFunc callback passed to the Upload call used for simple datatx put requests
+// * AFAICT only search and audir consume the path.
+//   - search needs to index from the root anyway. and it only needs the most recent path to put it in the index
+//   - audit on the other hand needs to log events with the path at the state of the event ... so it does need the full path.
+//     I think we can safely read determine the path later, right before emitting the event. and maybe make it configurable, because only audit needs it, anyway.
+func (s *OcisSession) Dir() string {
+	return s.info.Storage["Dir"]
+}
+
+func (s *OcisSession) Size() int64 {
+	return s.info.Size
+}
+func (s *OcisSession) SizeDiff() int64 {
+	sizeDiff, _ := strconv.ParseInt(s.info.MetaData["sizeDiff"], 10, 64)
+	return sizeDiff
+}
+
+func (s *OcisSession) ResourceID() provider.ResourceId {
```

Comment:
Exported func `ResourceID` should have a comment or be unexported

Reply:
removed unused code